src/solc_0.8/defi/ERC20RewardPool.sol

```diff
@@ -0,0 +1,428 @@
+        external
+        nonReentrant
+        antiDepositCheck(_msgSender())
+        checkRequirement(_msgSender(), amount, _balances[_msgSender()])
+    {
+        require(amount > 0, "ERC20RewardPool: Cannot stake 0");
+
+        // The first time a user stakes he cannot remove his rewards immediately.
+        if (timeLockClaim.lastClaim[_msgSender()] == 0) {
+            timeLockClaim.lastClaim[_msgSender()] = block.timestamp;
+        }
+
+        lockDeposit.lastDeposit[_msgSender()] = block.timestamp;
+
+        uint256 earlierRewards = 0;
+
+        if (_totalContributions == 0 && rewardCalculator != IRewardCalculator(address(0))) {
+            earlierRewards = rewardCalculator.getRewards();
+        }
+
+        _processRewards(_msgSender());
+        super._stake(amount);
+        _updateContribution(_msgSender());
+        require(_contributions[_msgSender()] > 0, "ERC20RewardPool: not enough contributions");
+
+        if (earlierRewards != 0) {
+            rewards[_msgSender()] = rewards[_msgSender()] + earlierRewards;
+        }
+        emit Staked(_msgSender(), amount);
+    }
+
+    /// @notice withdraw the stake from the contract
+    /// @param amount the amount of tokens to withdraw
+    /// @dev the user can withdraw his stake independently from the rewards
+    function withdraw(uint256 amount) external nonReentrant {
+        _processRewards(_msgSender());
+        _withdrawStake(_msgSender(), amount);
+        _updateContribution(_msgSender());
+    }
+
+    /// @notice withdraw the stake and the rewards from the contract
+    function exit() external nonReentrant {
+        _processRewards(_msgSender());
+        _withdrawStake(_msgSender(), _balances[_msgSender()]);
+        _withdrawRewards(_msgSender());
+        _updateContribution(_msgSender());
+        emit Exit(_msgSender());
+    }
+
+    /// @notice withdraw the rewards from the contract
+    /// @dev the user can withdraw his stake independently from the rewards
+    function getReward() external nonReentrant {
+        _processRewards(_msgSender());
+        _withdrawRewards(_msgSender());
+        _updateContribution(_msgSender());
+    }
+
+    function _withdrawStake(address account, uint256 amount) internal antiWithdrawCheck(_msgSender()) {
+        require(amount > 0, "ERC20RewardPool: Cannot withdraw 0");
+        super._withdraw(amount);
+        lockWithdraw.lastWithdraw[_msgSender()] = block.timestamp;
+        emit Withdrawn(account, amount);
+    }
+
+    function _withdrawRewards(address account) internal timeLockCheck(account) {
+        uint256 reward = rewards[account];
+        if (reward > 0) {
+            if (amountLockClaim.claimLockEnabled == true && amountLockClaim.amount > 0) {
+                // constrain the reward amount to the integer allowed
+                uint256 mod = reward % DECIMALS_18;
+                reward = reward - mod;
+                require(
+                    amountLockClaim.amount > reward,
```

Comment:
reverse

Reply:
amountLockClaim.amount = 0
amountLockClaim.claimLockEnabled = true
reward = 1.5
reward of 1.5 instead of 1